crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -257,7 +257,9 @@ impl C8YHttpProxyActor {
         self.get_fresh_token().await?;
         // build the request
         let request_builder = build_request(&self.end_point);
-        let request = request_builder?.build()?;
+        let request = request_builder?
+            .bearer_auth(self.end_point.token.clone().unwrap_or_default())
+            .build()?;
```

Comment:
If  `self.get_fresh_token().await?` where returning a `&Token` explicitly passed to the build request method, then things would be simpler to get correct:

```
let token = self.get_fresh_token().await?;
let request_builder = build_request(&self.end_point, token);
let request = request_builder?.build()?;
```